pantab/src/reader.c

```diff
@@ -152,11 +159,22 @@ PyObject *read_hyper_query(PyObject *Py_UNUSED(dummy), PyObject *args) {
       break; // No more to parse
     }
 
-    hyper_err = hyper_rowset_chunk_field_values(chunk, &num_cols, &num_rows,
-                                                &values, &sizes, &null_flags);
-
-    if (hyper_err) {
-      goto ERROR_CLEANUP;
+    if (hyper14567_compat) {
+      // Some trickery to support different APIs
+      // See
+      // https://stackoverflow.com/questions/71630729/provide-compatability-for-various-function-signatures-from-shared-library
+      ((void (*)(struct hyper_rowset_chunk_t *, size_t *, size_t *,
```

Comment:
As far as I can tell the return type being different here will be problematic. This appears to be undefined behavior

https://stackoverflow.com/questions/58548331/is-it-legal-to-cast-a-function-returning-an-object-pointer-to-a-function-returni